File: backend/src/main/java/de/bund/digitalservice/ris/caselaw/domain/Source.java
package de.bund.digitalservice.ris.caselaw.domain;

import java.util.Optional;
import lombok.Builder;
import org.jetbrains.annotations.NotNull;

@Builder(toBuilder = true)
public record Source(SourceValue value, String sourceRawValue, Reference reference) {
  @NotNull
  @Override
  public String toString() {
    return Optional.ofNullable(value).map(SourceValue::getLabel).orElse(sourceRawValue);
  }
}

File: frontend/src/domain/source.ts
import Reference from "@/domain/reference"

export enum SourceValue {
  UnaufgefordertesOriginal = "O",
  AngefordertesOriginal = "A",
  Zeitschrift = "Z",
  Email = "E",
  LaenderEuGH = "L",
  Sonstige = "S",
}

export type Source = {
  value?: SourceValue
  reference?: Reference
  sourceRawValue?: string
}
